bql/planner/planner.go

```diff
@@ -788,6 +789,204 @@ func (p *queryPlan) String() string {
 	return b.String()
 }
 
+// constructPlan encapsulates the sequence of instructions that need to be
+// executed in order to satisfy the execution of a valid construct BQL statement.
+type constructPlan struct {
+	stm       *semantic.Statement
+	store     storage.Store
+	tracer    io.Writer
+	queryPlan *queryPlan
+}
+
+func (p *constructPlan) Execute(ctx context.Context) (*table.Table, error) {
+	tbl, err := p.queryPlan.Execute(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var ts []*triple.Triple
+	for _, cc := range p.stm.ConstructClauses() {
+		for _, r := range tbl.Rows() {
+			sbj, prd, obj := cc.S, cc.P, cc.O
```

Comment:
Done. I can refactor this further when I refactor the construct and reification clauses into a single struct in a follow-up PR.